lib/SyTest/Federation/Room.pm

```diff
@@ -0,0 +1,138 @@
+package SyTest::Federation::Room;
+
+use strict;
+use warnings;
+
+sub make_event_refs
+{
+   [ map { [ $_->{event_id}, $_->{hashes} ] } @_ ];
```

Comment:
That structure is part of Matrix federation protocol itself. The protocol says that the `prev_events` and `auth_events` are lists of 2-element lists, each element giving an event ID and a hash.